cpp/include/cudf/text/detail/tokenize.hpp

```diff
@@ -0,0 +1,162 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+
+namespace cudf
+{
+namespace nvtext
+{
+namespace detail
+{
+
+/**
+ * @brief Returns a single column of strings by tokenizing the input strings
+ * column using the provided characters as delimiters.
+ *
+ * The `delimiter` may be zero or more characters. If the `delimiter` is empty,
+ * whitespace (character code-point <= ' ') is used for identifying tokens.
+ * Also, any consecutive delimiters found in a string are ignored.
+ * This means only non-empty tokens are returned.
+ *
+ * Tokens are found by locating delimiter(s) starting at the beginning of each string.
+ * As each string is tokenized, the tokens are appended using input column row order
+ * to build the output column. That is, tokens found in input row[i] will be placed in
+ * the output column directly before tokens found in input row[i+1].
+ *
+ * Example:
+ * ```
+ * s = ["a", "b c", "d  e f "]
+ * t = tokenize(s)
+ * t is now ["a","b","c","d","e","f"]
+ * ```
+ *
+ * All null row entries are ignored and the output contains all valid rows.
+ *
+ * @param strings Strings column tokenize.
+ * @param delimiter UTF-8 characters used to separate each string into tokens.
+ *                  The default of empty string will separate tokens using whitespace.
+ * @param mr Resource for allocating device memory.
+ * @param stream Stream to use for any CUDA calls.
+ * @return New strings columns of tokens.
+ */
+std::unique_ptr<column> tokenize( strings_column_view const& strings,
+                                  string_scalar const& delimiter = string_scalar{""},
+                                  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                  cudaStream_t stream = 0 );
+
+/**
+ * @brief Returns a single column of strings by tokenizing the input strings
+ * column using multiple strings as delimiters.
+ *
+ * Tokens are found by locating delimiter(s) starting at the beginning of each string.
+ * Any consecutive delimiters found in a string are ignored.
+ * This means only non-empty tokens are returned.
+ *
+ * As each string is tokenized, the tokens are appended using input column row order
+ * to build the output column. That is, tokens found in input row[i] will be placed in
+ * the output column directly before tokens found in input row[i+1].
+ *
+ * Example:
+ * ```
+ * s = ["a", "b c", "d,e:f;"]
+ * d = [",",":",";"]
```

Comment:
Some spaces make reading this easier.
```suggestion
 * d = [",", ":", ";"]
```